src/dd/dd.rs

```diff
@@ -0,0 +1,745 @@
+    pub fn set_flag(&mut self, flag: FlagT) {
+        self.flags |= flag;
+    }
+
+    pub fn get_flag(flag: &str) -> FlagT {
+        match flag {
+            "block" => Conf::BLOCK,
+            "unblock" => Conf::UNBLOCK,
+            "lcase" => Conf::LCASE,
+            "ucase" => Conf::UCASE,
+            "swab" => Conf::SWAB,
+            "noerror" => Conf::NOERROR,
+            "notrunc" => Conf::NOTRUNC,
+            "sync" => Conf::SYNC,
+            _ => Conf::INVALID_FLAG,
+        }
+    }
+
+    pub fn has_flag(&self, flag: FlagT) -> bool {
+        if self.flags & flag != 0 {
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn str_flags(&self) -> String {
+        let mut result = String::new();
+        if self.flags & Self::BLOCK != 0 {
+            result += "[block]";
+        }
+        if self.flags & Self::UNBLOCK != 0 {
+            result += "[unblock]";
+        }
+        if self.flags & Self::LCASE != 0 {
+            result += "[lcase]";
+        }
+        if self.flags & Self::UCASE != 0 {
+            result += "[ucase]";
+        }
+        if self.flags & Self::SWAB != 0 {
+            result += "[swab]";
+        }
+        if self.flags & Self::NOERROR != 0 {
+            result += "[noerror]";
+        }
+        if self.flags & Self::NOTRUNC != 0 {
+            result += "[notrunc]";
+        }
+        if self.flags & Self::SYNC != 0 {
+            result += "[sync]";
+        }
+        result
+    }
+
+    const BLOCK: FlagT = 0b00000001;
+    const UNBLOCK: FlagT = 0b00000010;
+    const LCASE: FlagT = 0b00000100;
+    const UCASE: FlagT = 0b00001000;
+    const SWAB: FlagT = 0b00010000;
+    const NOERROR: FlagT = 0b00100000;
+    const NOTRUNC: FlagT = 0b01000000;
+    const SYNC: FlagT = 0b10000000;
+    const INVALID_FLAG: FlagT = 0b1_00000000;
+}
+
+impl fmt::Display for Conf {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "if: {if_}\n\
+             of: {of_}\n\
+             ibs: {ibs}\n\
+             obs: {obs}\n\
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            if_ = self.inp_name,
+            of_ = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "ibs" => if let Some(m) = get_u(val) {
```

Comment:
You might want to just create another `match` for all of these options that basically have the same code (to reduce the boilerplate).  So maybe do something like:
```rust
match key {
    "if" => cf.set_if(val),
    "of" => cf.set_of(val),
    "conv" => {
        for elem in val.split(",") {
            match Conf::get_flag(elem) {
                Conf::INVALID_FLAG => err("invalid flag passed to conv"),
                flag => cf.set_flag(flag),
            }
        }
    }
    _ => {
        if let Some(m) = get_u(val) {
            match key {
                "ibs" => cf.inp_block_size = m,
                "obs" => cf.out_block_size = m,
                "bs" => {
                    cf.inp_block_size = m;
                    cf.out_block_size = m;
                }
                "cbs" => cf.cnv_block_size = m,
                "skip" => cf.skip = m,
                "seek" => cf.seek = m,
                "count" => cf.count = m
                _ => err(format!("invalid argument: {}", key))
            }
        } else {
            err(format!("invalid {}: {}", key, val));
        }
    }
}
```

Although doing something like the above would mostly likely slow down the code as you are now matching twice rather than once.

Reply:
Speed doesn't matter, as this small amount of arguments is parsed only once, maintanability is a way more important. Thanks
for useful tip.

Reply:
It's logical error here: value is checked before checking that key exists.

Reply:
I don't think it matters either way as the GNU version gives the same error in both cases (it's like "invalid operand: key=value" or something).